File: runtime/cpp/titan-ast-runtime-lib/DerivedTerminalGrammarAutomataData.cpp
//
// Created by tian wei jun on 2024/1/9.
//

#include "DerivedTerminalGrammarAutomataData.h"

RootTerminalGrammarMap::RootTerminalGrammarMap(
    Grammar *rootTerminalGrammar,
    std::unordered_map<std::string *, Grammar *, TextTerminalMapHash,
                       TextTerminalMapEq>
        textTerminalMap)
    : rootTerminalGrammar(rootTerminalGrammar),
      textTerminalMap(textTerminalMap) {}
RootTerminalGrammarMap::~RootTerminalGrammarMap() = default;

DerivedTerminalGrammarAutomataData::DerivedTerminalGrammarAutomataData()
    : count(0), rootTerminalGrammarMaps(std::vector<RootTerminalGrammarMap>()) {}

DerivedTerminalGrammarAutomataData::~DerivedTerminalGrammarAutomataData() =
    default;

File: runtime/java/titan-ast-runtime/src/titan/ast/runtime/DerivedTerminalGrammarAutomataData.java
package titan.ast.runtime;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

/**
 * .
 *
 * @author tian wei jun
 */
class DerivedTerminalGrammarAutomataData {

  int count = 0;
  List<RootTerminalGrammarMap> rootTerminalGrammarMaps = new ArrayList<>();

  static class RootTerminalGrammarMap {
    Grammar rootTerminalGrammar;
    HashMap<String, Grammar> textTerminalMap;

    public RootTerminalGrammarMap(
        Grammar rootTerminalGrammar, HashMap<String, Grammar> textTerminalMap) {
      this.rootTerminalGrammar = rootTerminalGrammar;
      this.textTerminalMap = textTerminalMap;
    }
  }
}
